### PR TITLE
Record C# 8.0 into language history

### PR DESCRIPTION
**Pull request by jcouv**

### CODE REVIEW

**Comment by ufcpp** [comment]
Also https://github.com/dotnet/csharplang/issues/1284

**Comment by jcouv** [comment]
FYI @agocke @MadsTorgersen @gafter @cston 
I'll merge tomorrow if no feedback

File: Language-Version-History.md [hunk 1]
```diff
@@ -104,3 +104,19 @@ Features Added in C# Language Versions
 - [Expression variables in initializers and queries](https://github.com/dotnet/csharplang/blob/master/proposals/csharp-7.3/expression-variables-in-initializers.md): Expression variables like `out var` and pattern variables are allowed in field initializers, constructor initializers and LINQ queries.
 -	[Tuple comparison](https://github.com/dotnet/csharplang/blob/master/proposals/csharp-7.3/tuple-equality.md): Tuples can now be compared with `==` and `!=`.
 -	[Attributes on backing fields](https://github.com/dotnet/csharplang/blob/master/proposals/csharp-7.3/auto-prop-field-attrs.md): Allows `[field: …]` attributes on an auto-implemented property to target its backing field.
+
+# C# 8.0 (Visual Studio 2019 version 16.3, .NET Core 3.0)
+- [Nullable reference types](https://github.com/dotnet/csharplang/blob/master/proposals/csharp-8.0/nullable-reference-types-specification.md): express nullability intent on reference types with `?`, `notnull` constraint and annotations attributes in APIs, the compiler will use those to try and detect possibly dangerous `null` values being dereferenced or passed to unsuitable APIs.
```
Comment (cston) [inline Language-Version-History.md]: >possibly dangerous [](start = 302, length = 18)

Consider removing.

Comment (cston) [inline Language-Version-History.md]: >- [](start = 0, length = 1)

Consider adding line breaks within list items.

File: Language-Version-History.md [hunk 2]
```diff
@@ -104,3 +104,19 @@ Features Added in C# Language Versions
...
+- [Default interface members](https://github.com/dotnet/csharplang/blob/master/proposals/csharp-8.0/default-interface-methods.md): interfaces can now have members can with default implementations, as well as static/private/protected/internal members except for state (ie. no fields).
```
Comment (cston) [inline Language-Version-History.md]: >can [](start = 163, length = 3)

Typo?

File: Language-Version-History.md [hunk 3]
```diff
@@ -104,3 +104,19 @@ Features Added in C# Language Versions
...
+- [Enhanced using](https://github.com/dotnet/csharplang/blob/master/proposals/csharp-8.0/using.md): a `using` declaration is added with an implicit scope and `using` statements and declarations allow disposal of `ref` structs using a pattern.
+- [Ranges and indexes](https://github.com/dotnet/csharplang/blob/master/proposals/csharp-8.0/ranges.md): the `i..j` syntax allows constructing `System.Range` instances, the `^k` syntax allows constructing `System.Index` instances, and those can be used to index/slice collections.
+- [Null-coalescing assignment](https://github.com/dotnet/csharplang/issues/34): `??=` allows conditionally assigning when the value is null.
+- [Static local function](https://github.com/dotnet/csharplang/issues/1565): local functions modified with `static` cannot capture `this` or local variables, and local function parameters now shadow locals in parent scopes.
```
Comment (cston) [inline Language-Version-History.md]: >function [](start = 16, length = 8)

"functions" (plural) to match the other entries.

File: Language-Version-History.md (hunk 1 above)
Comment (jcouv) [reply Language-Version-History.md]: Adding line breaks messes up the final rendering

---
In reply to: [305054988](https://github.com/dotnet/csharplang/pull/2666#discussion_r305054988) [](ancestors = 305054988)

**Pull request merged**